### PR TITLE
Periodical refresh of app token

### PR DESCRIPTION
**Pull request by Cloud11PL**
I want to merge this change because:
* Adds a token refresh method for app tokens
* Removed deprecated app settings views
* ~~Updates package-lock to v2~~

**PR intended to be tested with API branch:** 

### Pull Request Checklist

1. [ ] This code contains UI changes
2. [ ] All visible strings are translated with proper context including data-formatting
3. [ ] Attributes `[data-test-id]` are added for new elements
4. [ ] Changes are mentioned in the changelog
5. [ ] The changes are tested in different browsers and in light/dark mode

### Test environment config

API_URI=https://automation-dashboard.staging.saleor.cloud/graphql/
MARKETPLACE_URL=https://marketplace-gray.vercel.app/

### CODE REVIEW

**Comment by patrys** [comment]
Task linked: [SALEOR-7005 Add periodical refresh of app token in dashboard](https://app.clickup.com/t/2549495/SALEOR-7005)

**Review by witoszekdev — COMMENTED** [review]
I had a few questions for `useTokenRefresh` hook

I'm also wondering - what would happen if user has multiple tabs open? Would it mean that the token is refreshed by multiple tabs and in turn makes other open tabs unable to communicate with API (token was refreshed). I may be overthinking this 😅

File: src/apps/components/AppFrame/useTokenRefresh.ts [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+  iat?: number;
+}
+
+const TIME_BEFORE_REFRESH = 30 * 1000; // 30 seconds
+
+const useTokenRefresh = (token?: string, refetch?: () => void) => {
+  let decoded: AppToken;
+
+  // For some reason jwt_decode causes seemingly unrelated error in tests
+  // It seems like at some point undefined token is passed
```
[the remark this answers is not in the document]
Comment (Cloud11PL) [reply src/apps/components/AppFrame/useTokenRefresh.ts:15]: Types indicate that token is never `undefined`, however for some reason when jest tests is run somehow (I think) `undefined` token is passed causing a `jwt_decode` error. I wrapped it in `if` and `try..catch` just for the sake of tests as in the browser is works perfectly fine without them and `decodedSuccessfully`.

**Review by jwm0 — COMMENTED** [review]
Bump the node version in package.json if we want to support lock v2 (node 16/npm7) by default

**Review by witoszekdev — APPROVED** [review]

**Review by krzysztofwolski — APPROVED** [review]

**Pull request merged**